@06kellyjac @ofalvai I fixed update script using nurl and nix-prefetch-url, nix-hash, but still has build error
```
error: failed to run custom build command for `libsqlite3-sys v0.30.1`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/libsqlite3-sys-377e06a57289c3e1/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=LIBSQLITE3_SYS_USE_PKG_CONFIG
  cargo:rerun-if-env-changed=LIBSQLITE3_SYS_BUNDLING

  --- stderr
  sqlite3/sqlite3.h:35:10: fatal error: 'stdarg.h' file not found
  thread 'main' panicked at /build/deno-2.2.1-vendor/libsqlite3-sys-0.30.1/build.rs:620:33:
  could not run bindgen on header sqlite3/sqlite3.h
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  ```

--- a/pkgs/by-name/de/deno/librusty_v8.nix
+++ b/pkgs/by-name/de/deno/librusty_v8.nix
@@ -2,11 +2,11 @@
 { fetchLibrustyV8 }:
 
 fetchLibrustyV8 {
-  version = "130.0.1";
+  version = "134.4.0";
   shas = {
-    x86_64-linux = "sha256-MKdqaIF3M7j9IraOezoP5szr+SDfTg0iUOEtwe76N7k=";
-    aarch64-linux = "sha256-qc25H3Aj2KRhsAZ+2SD1c4RmweVK07oW71opZXRuUoc=";
-    x86_64-darwin = "sha256-CYw4AdBjFXmRqkTcxqT9Z0n0rBTxUsqrHxcf8c9Q8ww=";
-    aarch64-darwin = "sha256-d1QTLt8gOUFxACes4oyIYgDF/srLOEk+5p5Oj1ECajQ=";
+    x86_64-linux = "sha256-yG9p8b0dQTtoa234eTRBumjQ5GWxWJFsKHbgNFCUUAA=";
+    aarch64-linux = "sha256-5IrviupPjTUHTeuwmeA3FHuOy2H8qc05WNqRmlE0UJk=";
+    x86_64-darwin = "sha256-RPEW5a51WjfYyPjvkKvFbs+fPTlsv9/2sFtqUdtz7EY=";
+    aarch64-darwin = "sha256-PFyJTOYHWw3YOoG85w+y0899vc0+psz0TtCMXORcsHI=";
   };
 }

--- a/pkgs/by-name/de/deno/package.nix
+++ b/pkgs/by-name/de/deno/package.nix
@@ -11,6 +11,7 @@
     inherit (callPackage ./fetchers.nix { }) fetchLibrustyV8;
   },
   libffi,
+  sqlite,
 }:
 
 let
@@ -18,17 +19,17 @@ let
 in
 rustPlatform.buildRustPackage rec {
   pname = "deno";
-  version = "2.1.10";
+  version = "2.2.1";
 
   src = fetchFromGitHub {
     owner = "denoland";
     repo = "deno";
     tag = "v${version}";
-    hash = "sha256-1NqBNrTYHhsG6O2qsgANM15FOxx3xheE9T3nYDDH1D0=";
+    hash = "sha256-WXAUsBC3nAJcuUB753dpM/WDzqWu+e/Kt/BrwQkk/dY=";
   };
 
   useFetchCargoVendor = true;
-  cargoHash = "sha256-D7dt9RdlTmR2C1dwOl5vZKUlLUnTkdudtZ4FQpr9ddU=";
+  cargoHash = "sha256-t44Q4yBdcYAk6jkRrzAHXBsJTsRTHAD95Wyxd3waaHc=";
 
   postPatch = ''
     # upstream uses lld on aarch64-darwin for faster builds
@@ -42,6 +43,7 @@ rustPlatform.buildRustPackage rec {
   # uses zlib-ng but can't dynamically link yet
   # https://github.com/rust-lang/libz-sys/issues/158
   nativeBuildInputs = [
+    rustPlatform.bindgenHook
     # required by libz-ng-sys crate
     cmake
     # required by deno_kv crate
@@ -54,7 +56,11 @@ rustPlatform.buildRustPackage rec {
     "--disable-multi-os-directory"
   ];
 
-  buildInputs = [ libffi ];
+  buildInputs = [
+    libffi
+    # required by libsqlite3-sys
+    sqlite.dev
+  ];
   buildAndTestSubdir = "cli";
 
   # work around "error: unknown warning group '-Wunused-but-set-parameter'"

--- a/pkgs/by-name/de/deno/update/librusty_v8.ts
+++ b/pkgs/by-name/de/deno/update/librusty_v8.ts
@@ -25,16 +25,12 @@ const fetchArchShaTasks = (version: string, arches: Architecture[]) =>
   arches.map(
     async (arch: Architecture): Promise<PrefetchResult> => {
       log("Fetching:", arch.nix);
-      const sha256 = await run("nix-prefetch", [
-        `
-{ fetchurl }:
-fetchurl {
-  url = "https://github.com/denoland/rusty_v8/releases/download/v${version}/librusty_v8_release_${arch.rust}.a.gz";
-}
-`,
+      const sha256 = await run("nix-prefetch-url", [
+        `https://github.com/denoland/rusty_v8/releases/download/v${version}/librusty_v8_release_${arch.rust}.a.gz`
       ]);
+      const sha256_sri = await run("nix-hash", ["--type", "sha256", "--to-sri", sha256]);
       log("Done:    ", arch.nix);
-      return { arch, sha256 };
+      return { arch, sha256: sha256_sri };
     },
   );
 
@@ -45,10 +41,6 @@ const templateDeps = (version: string, deps: PrefetchResult[]) =>
 fetchLibrustyV8 {
   version = "${version}";
   shas = {
-    x86_64-linux = "sha256-jLYl/CJp2Z+Ut6qZlh6u+CtR8KN+ToNTB+72QnVbIKM=";
-    aarch64-linux = "sha256-uAkBMg6JXA+aILd8TzDtuaEdM3Axiw43Ad5tZzxNt5w=";
-    x86_64-darwin = "sha256-60aR0YvQT8KyacY8J3fWKZcf9vny51VUB19NVpurS/A=";
-    aarch64-darwin = "sha256-pd/I6Mclj2/r/uJTIywnolPKYzeLu1c28d/6D56vkzQ=";
 ${deps.map(({ arch, sha256 }) => `    ${arch.nix} = "${sha256}";`).join("\n")}
   };
 }

--- a/pkgs/by-name/de/deno/update/src.ts
+++ b/pkgs/by-name/de/deno/update/src.ts
@@ -16,11 +16,11 @@ interface Replacer {
 const log = logger("src");
 
 const prefetchHash = (nixpkgs: string, version: string) =>
-  run("nix-prefetch", ["-f", nixpkgs, "deno.src", "--rev", version]);
+  run("nurl", ["https://github.com/denoland/deno", version, "-H", "-n", nixpkgs]);
 const prefetchCargoHash = (nixpkgs: string) =>
   run(
-    "nix-prefetch",
-    [`{ sha256 }: (import ${nixpkgs} {}).deno.cargoDeps.overrideAttrs (_: { hash = sha256; })`],
+    "nurl",
+    ["-e", `(import ${nixpkgs} {}).deno.cargoDeps.overrideAttrs (_: { hash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="; })`]
   );
 
 const replace = (str: string, replacers: Replacer[]) =>

--- a/pkgs/by-name/de/deno/update/update.ts
+++ b/pkgs/by-name/de/deno/update/update.ts
@@ -1,6 +1,6 @@
 #!/usr/bin/env nix-shell
 /*
-#!nix-shell -i "deno run --allow-net --allow-run --allow-read --allow-write" -p deno git nix-prefetch
+#!nix-shell -i "deno run --allow-net --allow-run --allow-read --allow-write" -p deno git nurl
 */
 import { getExistingVersion, getLatestVersion, logger } from "./common.ts";
 import { Architecture, updateLibrustyV8 } from "./librusty_v8.ts";

